src/MensaWidget/MWFrameTest.class.st

```diff
@@ -36,6 +36,24 @@ MWFrameTest >> tearDown [
 	widgetFrame abandon
 ]
 
+{
+	#category : #tests,
+	#'squeak_changestamp' : 'tmc 6/6/2024 22:21'
+}
+MWFrameTest >> testCantineCount [
+
+	self widgetFrame submorphsDo: [ :each | 
+		each class = ScrollPane ifTrue: 
+			[ each submorphsDo: [ :each2 |
+				each2 class = TransformationMorph ifTrue: 
+					[ each2 submorphsDo: [ :each3 |
+						each3 class = AlignmentMorph ifTrue: [
+							self assert: each3 subclasses size = 
+								MWDataGettingWrapper canteenDict size]]]]]]
+				 
+			  
```

Comment:
as stated above and is there a way to extract some code or at least a loop into an extra method? i think it's not very readable